model/Delete/DeliveryDeleteService.php

```diff
@@ -216,13 +244,35 @@ private function buildDeliveryExecutionDeleteRequest($deliveryResource, $executi
         );
     }
 
+    private function deleteLinkedResources(): void
+    {
+        $delivery = $this->request->getDeliveryResource();
+
+        if (!$delivery->exists()) {
+            return;
+        }
+
+        $test = $delivery->getUniquePropertyValue(
+            $delivery->getProperty(DeliveryAssemblyService::PROPERTY_ORIGIN)
+        );
+
+        $itemService = $this->getItemService();
+
+        foreach ($this->getQtiTestService()->getItems($test) as $item) {
+            $itemService->deleteResource($item);
+        }
+
+        $this->getTestService()->deleteResource($test);
+    }
+
     /**
-     * @param core_kernel_classes_Resource $delivery
-     * @param $executions
+     * @param DeliveryExecution[] $executions
      * @throws \common_exception_Error
      */
-    private function deleteDeliveryExecutions(\core_kernel_classes_Resource $delivery, $executions)
+    private function deleteDeliveryExecutions(array $executions): void
```

Comment:
by changing this contract you risk to break the backwards  compatibility

Reply:
On a private function? I don't think so ;)